pkg/controller/appinst_api.go

```diff
@@ -1866,11 +1936,38 @@ func (s *AppInstApi) deleteAppInstInternal(cctx *CallContext, in *edgeproto.AppI
 		reqCtx, reqCancel := context.WithTimeout(ctx, s.all.settingsApi.Get().DeleteAppInstTimeout.TimeDuration())
 		defer reqCancel()
 
-		err = edgeproto.WaitForAppInstInfo(reqCtx, &in.Key, s.store, edgeproto.TrackedState_NOT_PRESENT,
-			DeleteAppInstTransitions, edgeproto.TrackedState_DELETE_ERROR,
-			"Deleted AppInst successfully", cb.Send,
-			edgeproto.WithCrmMsgCh(sendObj.crmMsgCh),
-		)
+		crmAction := func() error {
```

Comment:
Same comment - why not inline the function, since it just gets called right away?